src/components/Product.tsx

```diff
@@ -0,0 +1,68 @@
+import { IData } from '../types/types';
+
+import { useProduct } from '../Context/ProductContextProvider';
+
+import React from 'react';
+import Card from '@mui/material/Card';
+import { Button, CardActions, CardContent, CardMedia, Typography } from '@mui/material';
+import AddShoppingCartIcon from '@mui/icons-material/AddShoppingCart';
+
+
+export const Product = (el: IData) => {
+
+  const { addProduct, removeProduct, products } = useProduct();
+
+  const addProductToCart = (el: IData) => {
+    addProduct(el);
+    el.handleClose();
+  };
+
+  const removeProductFromCart = (id: number) => {
+    removeProduct(id);
+  };
+
+  const checkProductInStorage = products.some((e) => e.id === el.id);
+
+  return (
+    <div style={{ height: '100%' }}>
```

Comment:
just small advice, you can use <Box> instead of div and just use sx={} there as well.
`<Box sx={...}> ... </Box>`